src/storage.rs

```diff
@@ -72,4 +83,87 @@ impl BeerTally for HashMapBeerTally {
             },
         }
     }
+
+    fn get_players_vect(&mut self, chat_id: i64) -> Result<Vec<tally::UserTally>, &str> { 
+        if let Some(chat) = self.players.get_mut(&chat_id) {
+            if chat.is_empty() {
+                // Found the chat, but no users are registered
+                // Someone probably unregistered
+                Err("No users have been initialized in this chat, try to register some users first")
+            } else {
+                Ok(chat.values().cloned().collect())
+            }
+        } else {
+            // The chat could not be found (no one has registered yet)
+            Err("No users have been registered yet")
+        }
+    }
+
+    fn get_player(&mut self, chat_id: i64, user_id: i64) -> Result<&mut tally::UserTally, &str> { 
+        if let Some(chat) = self.players.get_mut(&chat_id) {
+            if chat.is_empty() {
+                // Found the chat, but no users are registered
+                // Someone probably unregistered
+                Err("No users have been initialized in this chat, try to register some users first")
+            } else if let Some(user) = chat.get_mut(&user_id) {
+                Ok(user)
+            } else {
+                // This should not happen
+                Err("The user was not found")
+            }
+        } else {
+            // The chat could not be found (no one has registered yet)
+            Err("No users have been registered yet")
+        }
+    }
+
+    fn player_list(&mut self, chat_id: i64) -> String {
+        match self.get_players_vect(chat_id) {
+            Ok(list_of_players) => {
+                let mut return_string : String = "Players:".to_string();
+                for player in &list_of_players {
+                    return_string = return_string + "\n" + &player.name;
+                }
+                format!("{}", return_string)
+            },
+            Err(msg) =>  format!("{}", msg)
+        }
+    }
+
+    fn get_random(&mut self, chat_id: i64) -> String {
+        match self.get_players_vect(chat_id) {
+            Ok(list_of_players) => {
+                let rng = rand::thread_rng().gen_range(0..list_of_players.len());
+                format!("{}", list_of_players[rng].name)
```

Comment:
Shouldn't this just be
```suggestion
                list_of_players[rng].name
```
?

Maybe you need a `clone()` or something like that, not sure.